sonorancad/core/client.lua

```diff
@@ -128,24 +128,16 @@ RegisterNetEvent('SonoranCAD::Core::InitBodycam', function()
 	if Config.bodycamEnabled then
 	print('Bodycam init')
 	-- Command to toggle bodycam on and off
-	RegisterCommand(Config.bodycamCommandToggle, function(source, args, rawCommand)
-		if Config.apiVersion < 4 then
-			errorLog('Bodycam is only enabled with Sonoran CAD Pro.')
-			TriggerEvent('chat:addMessage', {
-				args = {
-					'Sonoran Bodycam',
-					'Bodycam is only enabled with Sonoran CAD Pro.'
-				}
-			})
-			return
-		end
-		if bodyCamOn then
-			bodyCamOn = false
-			TriggerServerEvent('SonoranCAD::core::bodyCamOff')
+	local function EnableBodyCam(enable)
```

Comment:
This function is overly complicated...it should get current state and flip the bodycam state. It could be a "ToggleBodyCam" function and be much less code.

Reply:
You are 100% right, I added a little bit of everything in there so it would be very flexible. Especially if called on twice. For example
BANG (trigger start bodycam with timer)
BANG (trigger start but its bypassed because the state is already on)
etc. 

It could be less lines of code but would require a check every time it happens. It could go either way sense both a method of getting the current state and a trigger state are defined.